src/coreclr/src/tools/Common/Compiler/VectorFieldLayoutAlgorithm.cs

```diff
@@ -106,8 +107,9 @@ public static bool IsVectorType(DefType type)
         {
```

Comment:
What will happen for say `Vector64<bool>` or `Vector128<Guid>` which are legal IL but unsupported by the JIT/runtime?

Reply:
Their alignment will be the same as for numeric `Vector64<T>` and `Vector128<T>` types. However, on ARM64 they will not be considered HFA/HVA and will be passed in X registers instead of Q registers. That is how VM and JIT work at present and Crossgen2 must be compatible.